### PR TITLE
Fix for visual waveform position while scratching outside of an activated loop

### LINKED ISSUE #12274: scratching waveform backwards with a saved, active loop ahead temporarily jumps to that loop (visuals only)
* create a hotcue loop
* jump back a few bars
* play
* ativate loop (catching loop)
* drag waveform forwards = visuals as expected
* drag waveform backwards = visual position jumps to loop (audio is not affected)

https://github.com/mixxxdj/mixxx/assets/5934199/0115009a-7463-4f45-a304-3de6acaa7db1

### Version

2.4-beta-216-g5454b94b71

### OS

Ubuntu Studio 20.04 / X11 / xfce 4.14 / Qt 5.12.8

### PR DESCRIPTION
**Pull request by JoergAtGithub**
Closes: #12274 
Add additional conditions to ensure, that loop boundary adjustment of the visual playposition is only done, when the engine playposition is inside the loop.

### CODE REVIEW

**Comment by ronso0** [comment]
Thanks for this quick fix, works as desired!

**Comment by ronso0** [comment]
the code LGTM locally, though I don't understand how it relates to drag'n'drop, I'd have to reverse-engineer too much, so someone else please take a look. Thanks.

File: src/waveform/visualplayposition.cpp [hunk 1]
```diff
@@ -88,13 +88,17 @@ double VisualPlayPosition::determinePlayPosInLoopBoundries(
     if (data.m_loopEnabled) {
         double loopSize = data.m_loopEndPos - data.m_loopStartPos;
         if (loopSize > 0) {
-            if ((data.m_playRate < 0.0) && (interpolatedPlayPos < data.m_loopStartPos)) {
+            if ((data.m_playRate < 0.0) &&
+                    (interpolatedPlayPos < data.m_loopStartPos) &&
+                    (data.m_playPos >= data.m_loopStartPos)) {
```
Comment (daschuer) [inline src/waveform/visualplayposition.cpp:93]: Please add a comment, when can this condition be false.
The same below.

**Review by daschuer — CHANGES_REQUESTED** [review]
Than you. The code looks good, but can make use of a comment.

File: src/waveform/visualplayposition.cpp (hunk 1 above)
Comment (JoergAtGithub) [reply src/waveform/visualplayposition.cpp:93]: Done!

**Review by daschuer — APPROVED** [review]
LGTM, Thank you for the detailed comment.

**Pull request merged**